### PR TITLE
fix: karma circle

### PR DESCRIPTION
**Pull request by yusufputra**
### Summary by CodeRabbit

- New Feature: Introduced a new `GradientCircularProgress` component for rendering gradient circular progress bars, enhancing the visual appeal of progress indicators.
- Refactor: Optimized the `CircleGradient` component by removing dependencies on external libraries and simplifying the code, improving performance.
- Style: Adjusted styles and properties of various user profile-related components including `DomainList`, `KarmaScore`, `ProfilePicture`, and `UserInfo`, enhancing the layout and visual appearance in the UI.
- Chore: Added a new color (`gray5`) to the theme color palette, providing more styling options.

### CODE REVIEW

**Comment by bastian-purrer** [comment]
Image description   CodeRabbit



Currently reviewing new changes in this PR...

<details>
<summary>Commits</summary>
Files that changed from the base of the PR and between 3e2623af7f44a97c0891e32fa9540f1fcd33f9f3 and 36001b15cf02cf6a00abfcd5a8eb9bb63f8fc11e commits.
</details>

<details>
<summary>Files selected (9)</summary>

* __mocks__/react-native-svg.tsx (1)
* src/components/Karma/CircleGradient.tsx (3)
* src/components/Karma/GradientCircularProgress.tsx (1)
* src/screens/DiscoveryScreenV2/elements/DiscoveryItemList.js (2)
* src/screens/OtherProfile/index.js (1)
* src/screens/ProfileScreen/elements/KarmaScore.tsx (1)
* src/screens/ProfileScreen/elements/ProfilePicture.js (2)
* src/screens/ProfileScreen/index.js (1)
* src/screens/WhotoFollow/elements/UserInfo.js (2)
</details>


<details>
<summary>Files ignored due to filter (1)</summary>

* __tests__/components/WhoToFollow/__snapshots__/WhoToFollowUserInfo.test.js.snap

</details>





---

File: src/screens/ProfileScreen/index.js [hunk 1]
```diff
@@ -103,7 +103,7 @@ const Header = (props) => {
           <ProfilePicture
             onImageContainerClick={changeImage}
             profilePicPath={dataMain.profile_pic_path}
-            karmaScore={Math.floor(dataMain.karma_score)}
+            karmaScore={dataMain.karma_score}
```
Comment (bastian-purrer) [inline src/screens/ProfileScreen/index.js:106]: Image description   CodeRabbit

The change from rounding down the `karmaScore` to passing it directly might cause issues if the `ProfilePicture` component or any of its child components expect an integer value for `karmaScore`. If fractional values can cause problems, consider keeping the `Math.floor()` function.
```diff
- karmaScore={Math.floor(dataMain.karma_score)}
+ karmaScore={dataMain.karma_score}
```
If the `ProfilePicture` component and its children can handle fractional `karmaScore` values without any issues, then this change is fine.

File: src/screens/OtherProfile/index.js [hunk 2]
```diff
@@ -642,7 +642,9 @@ const styles = StyleSheet.create({
   profileImage: {
     width: 95,
     height: 95,
-    borderRadius: 100
+    borderRadius: 100,
+    marginLeft: 3.2,
+    marginTop: 3.2
```
Comment (bastian-purrer) [inline src/screens/OtherProfile/index.js:647]: Image description   CodeRabbit

The changes made to the `profileImage` style are fine, but it's important to note that the `marginLeft` and `marginTop` values are in pixels. If you want to support multiple screen sizes or densities, consider using a relative unit like percentage or `react-native`'s `Dimensions` API to calculate these values based on the screen size.

```diff
-    marginLeft: 3.2,
-    marginTop: 3.2
+    marginLeft: Dimensions.get('window').width * 0.01, // for example, 1% of screen width
+    marginTop: Dimensions.get('window').height * 0.01, // for example, 1% of screen height
```
This way, the margin will scale appropriately on different screen sizes.

File: src/screens/ProfileScreen/elements/ProfilePicture.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ const ProfilePicture = ({
   return (
     <View style={styles.wrapImageProfile}>
       <TouchableNativeFeedback onPress={onImageContainerClick}>
-        <CircleGradient fill={karmaScore} size={normalize(100)} width={normalize(2.2)}>
+        <CircleGradient fill={karmaScore} size={normalize(100)} width={normalize(6)}>
```
Comment (bastian-purrer) [inline src/screens/ProfileScreen/elements/ProfilePicture.js:27]: Image description   CodeRabbit

The width of the `CircleGradient` component has been increased from `normalize(2.2)` to `normalize(6)`. This change will make the circular border around the profile picture thicker. Ensure this is the intended visual effect and that it doesn't obscure important parts of the profile image or clash with other UI elements.

```diff
- <CircleGradient fill={karmaScore} size={normalize(100)} width={normalize(2.2)}>
+ <CircleGradient fill={karmaScore} size={normalize(100)} width={normalize(6)}>
```

File: src/screens/ProfileScreen/elements/ProfilePicture.js [hunk 4]
```diff
@@ -44,12 +44,11 @@ const ProfilePicture = ({
...
-    width: 97,
-    height: 97,
+    width: dimen.normalizeDimen(92),
+    height: dimen.normalizeDimen(92),
     borderRadius: 100,
-    marginLeft: 1.4,
-    marginTop: 2,
-    marginBottom: 2
+    marginLeft: dimen.normalizeDimen(2.5),
+    marginTop: dimen.normalizeDimen(2.75)
```
Comment (bastian-purrer) [inline src/screens/ProfileScreen/elements/ProfilePicture.js:51]: Image description   CodeRabbit

The width and height of the `profileImage` style have been reduced, and the `marginLeft` and `marginTop` properties have been increased. These changes will result in a smaller profile image with more margin on the left and top. Make sure these changes align with your design specifications and don't cause any unexpected layout shifts.

```diff
-    width: 97,
-    height: 97,
-    marginLeft: 1.4,
-    marginTop: 2,
+    width: dimen.normalizeDimen(92),
+    height: dimen.normalizeDimen(92),
+    marginLeft: dimen.normalizeDimen(2.5),
+    marginTop: dimen.normalizeDimen(2.75)
```

**Review by bastian-purrer — COMMENTED** [review]
Image description   CodeRabbit

<details>
<summary>Commits</summary>
Files that changed from the base of the PR and between 3e2623af7f44a97c0891e32fa9540f1fcd33f9f3 and 36001b15cf02cf6a00abfcd5a8eb9bb63f8fc11e commits.
</details>

<details>
<summary>Files selected (9)</summary>

* __mocks__/react-native-svg.tsx (1)
* src/components/Karma/CircleGradient.tsx (3)
* src/components/Karma/GradientCircularProgress.tsx (1)
* src/screens/DiscoveryScreenV2/elements/DiscoveryItemList.js (2)
* src/screens/OtherProfile/index.js (1)
* src/screens/ProfileScreen/elements/KarmaScore.tsx (1)
* src/screens/ProfileScreen/elements/ProfilePicture.js (2)
* src/screens/ProfileScreen/index.js (1)
* src/screens/WhotoFollow/elements/UserInfo.js (2)
</details>


<details>
<summary>Files ignored due to filter (1)</summary>

* __tests__/components/WhoToFollow/__snapshots__/WhoToFollowUserInfo.test.js.snap

</details>






<details>
<summary>Files skipped from review due to trivial changes (6)</summary>

* __mocks__/react-native-svg.tsx
* src/components/Karma/CircleGradient.tsx
* src/components/Karma/GradientCircularProgress.tsx
* src/screens/DiscoveryScreenV2/elements/DiscoveryItemList.js
* src/screens/ProfileScreen/elements/KarmaScore.tsx
* src/screens/WhotoFollow/elements/UserInfo.js

</details>

<details>
<summary>Review comments generated (4)</summary>

* Review: 4
* LGTM: 0

</details>

---

<details>
<summary>Tips</summary>

### Chat with Image description  CodeRabbit Bot (`@coderabbitai`)
- Reply on review comments left by this bot to ask follow-up questions. A review comment is a comment on a diff or a file.
- Invite the bot into a review comment chain by tagging `@coderabbitai` in a reply.

### Code suggestions
- The bot may make code suggestions, but please review them carefully before committing since the line number ranges may be misaligned. 
- You can edit the comment made by the bot and manually tweak the suggestion if it is slightly off.

### Pausing incremental reviews
- Add `@coderabbitai: ignore` anywhere in the PR description to pause further reviews from the bot.

</details>

File: src/components/Karma/CircleGradient.tsx [hunk 5]
```diff
@@ -33,40 +23,21 @@ export const CircleGradient = (props: {
...
-        }}>
-        {props.children}
-      </LinearGradient>
-      {props.fill !== 100 && (
-        <Svg
-          height={props.size + Math.ceil(props.width)}
-          width={props.size + Math.ceil(props.width)}
-          viewBox={`0 0 ${props.size} ${props.size}`}
+      <GradientCircularProgress
+        emptyColor="#E8EBED"
```
Comment (asaadam) [inline src/components/Karma/CircleGradient.tsx:27]: nitip di pindah ke colors

Comment (yusufputra) [reply src/components/Karma/CircleGradient.tsx]: sydaa

**Review by bastian-purrer — COMMENTED** [review]
Image description   CodeRabbit

<details>
<summary>Commits</summary>
Files that changed from the base of the PR and between 36001b15cf02cf6a00abfcd5a8eb9bb63f8fc11e and 1b7507ead4ae0c2ee0e6898189217eef05d6417f commits.
</details>

<details>
<summary>Files selected (2)</summary>

* src/components/Karma/CircleGradient.tsx (3)
* src/utils/theme/index.js (1)
</details>







<details>
<summary>Files skipped from review due to trivial changes (2)</summary>

* src/components/Karma/CircleGradient.tsx
* src/utils/theme/index.js

</details>

<details>
<summary>Review comments generated (0)</summary>

* Review: 0
* LGTM: 0

</details>

---

<details>
<summary>Tips</summary>

### Chat with Image description  CodeRabbit Bot (`@coderabbitai`)
- Reply on review comments left by this bot to ask follow-up questions. A review comment is a comment on a diff or a file.
- Invite the bot into a review comment chain by tagging `@coderabbitai` in a reply.

### Code suggestions
- The bot may make code suggestions, but please review them carefully before committing since the line number ranges may be misaligned. 
- You can edit the comment made by the bot and manually tweak the suggestion if it is slightly off.

### Pausing incremental reviews
- Add `@coderabbitai: ignore` anywhere in the PR description to pause further reviews from the bot.

</details>

**Review by asaadam — APPROVED** [review]

**Pull request merged**